src/ck-core/charm++.h

```diff
@@ -942,66 +921,46 @@ class CProxyElement_Group : public CProxy_Group {
 #define GROUP_SECTION_PROXY 1
 class CProxySection_Group : public CProxy_Group {
 private:
-  int _nsid;
-  CkSectionID *_sid;
+  std::vector<CkSectionID> _sid;
 public:
   CProxySection_Group() { }
   CProxySection_Group(const CkGroupID &gid, const int *elems, const int nElems, int factor=USE_DEFAULT_BRANCH_FACTOR)
-      :CProxy_Group(gid), _nsid(1) { _sid = new CkSectionID(gid, elems, nElems, factor); }
+      :CProxy_Group(gid), _sid({CkSectionID(gid, elems, nElems, factor)}) { }
   CProxySection_Group(const CkGroupID &gid, const int *elems, const int nElems, CK_DELCTOR_PARAM)
-      :CProxy_Group(gid,CK_DELCTOR_ARGS), _nsid(1) { _sid = new CkSectionID(gid, elems, nElems); }
+      :CProxy_Group(gid,CK_DELCTOR_ARGS), _sid({CkSectionID(gid, elems, nElems)}) { }
   CProxySection_Group(const CProxySection_Group &cs)
-      :CProxy_Group(cs.ckGetGroupID()), _nsid(cs._nsid) {
-    if (_nsid == 1) _sid = new CkSectionID(cs.ckGetGroupID(), cs.ckGetElements(), cs.ckGetNumElements());
-    else if (_nsid > 1) {
-      _sid = new CkSectionID[_nsid];
-      for (int i=0; i<_nsid; ++i) _sid[i] = cs._sid[i];
-    } else _sid = NULL;
-  }
+      :CProxy_Group(cs.ckGetGroupID()), _sid(cs._sid) { }
   CProxySection_Group(const CProxySection_Group &cs,CK_DELCTOR_PARAM)
-      :CProxy_Group(cs.ckGetGroupID(),CK_DELCTOR_ARGS), _nsid(cs._nsid) {
-    if (_nsid == 1) _sid = new CkSectionID(cs.ckGetGroupID(), cs.ckGetElements(), cs.ckGetNumElements());
-    else if (_nsid > 1) {
-      _sid = new CkSectionID[_nsid];
-      for (int i=0; i<_nsid; ++i) _sid[i] = cs._sid[i];
-    } else _sid = NULL;
-  }
+      :CProxy_Group(cs.ckGetGroupID(),CK_DELCTOR_ARGS), _sid(cs._sid) { }
   CProxySection_Group(const IrrGroup *g)
-      :CProxy_Group(g), _nsid(0) {}
+      :CProxy_Group(g) { }
   CProxySection_Group(const int n, const CkGroupID *gid,  int const * const *elems, const int *nElems, int factor=USE_DEFAULT_BRANCH_FACTOR)
-      :CProxy_Group(gid[0]), _nsid(n) {
-    _sid = new CkSectionID[n];
-    for (int i=0; i<n; ++i) _sid[i] = CkSectionID(gid[i], elems[i], nElems[i], factor);
+      :CProxy_Group(gid[0]) {
+    _sid.reserve(n);
```

Comment:
It would be better to construct the vector to size or use resize(), then assign directly, rather than using emplace_back() iteratively which has to update the vector's back ptr each time.

Reply:
Done